### PR TITLE
Fixed issue#2

### PR DESCRIPTION
**Pull request by hritikgupta**
This PR solves the issue#2 of failing to decompress empty file.

### CODE REVIEW

File: src/srb/decompressor.cpp [hunk 1]
```diff
@@ -71,15 +71,20 @@ namespace srb
             { { 80, 75, 3, 4 }, ZIP },
             { { 120, -100 }, ZLIB }
         };
-
+        
+	/*
+	* If the first line of the VCF file has less than 5 characters,
```
Comment (srbcheema1) [inline src/srb/decompressor.cpp]: VCF xD,  this tool is not only for vcf files. that was vcf-validator which was for vcf files. it is made generic for all kind of files.

File: src/srb/decompressor.cpp [hunk 2]
```diff
@@ -71,15 +71,20 @@ namespace srb
             { { 80, 75, 3, 4 }, ZIP },
             { { 120, -100 }, ZLIB }
         };
-
+        
+	/*
+	* If the first line of the VCF file has less than 5 characters, 
+	* std::equal will cause a segmentation fault.
+	*/
+	if(line.size()<5)
```
Comment (srbcheema1) [inline src/srb/decompressor.cpp]: use proper spacing `if (line.size() < 5)`

File: src/srb/decompressor.cpp [hunk 3]
```diff
@@ -71,15 +71,20 @@ namespace srb
...
+		return NO_EXT;
```
Comment (srbcheema1) [inline src/srb/decompressor.cpp]: always use parentheses they are really helpful in future.
```
if (line.size() < 5) {
    return NO_EXT;
}
```

**Review by srbcheema1 — CHANGES_REQUESTED** [review]

File: src/srb/decompressor.cpp (hunk 1 above)
Comment (hritikgupta) [reply src/srb/decompressor.cpp]: Ah I see xD

File: src/srb/decompressor.cpp [hunk 4]
```diff
@@ -71,15 +71,21 @@ namespace srb
...
         };
-
+	
+	/*
+	* If the first line of the file has less than 5 characters,
+	* std::equal will cause a segmentation fault.
+	*/
+	if (line.size() < 5) {
+	    return NO_EXT;
+	}
```
Comment (srbcheema1) [inline src/srb/decompressor.cpp]: intendation is still wrong.

**Review by srbcheema1 — CHANGES_REQUESTED** [review]

**Comment by srbcheema1** [comment]
@hritikgupta you are going well. just a small change I would like to suggest. 
it would be even better to move the condition in for loop. checking if the size of line should be greater than type.siz() to do the std::equal check

```
        for (auto & type : types) {
            if (line.size() >= type.size()) {
                if (std::equal(type.first.begin(), type.first.end(), line.begin())) {
                    std::cout << "detected " << type.second
                        << " compression" << std::endl;
                    return type.second;
                }
            }
        }
```

**Comment by srbcheema1** [comment]
sorry my bad. I made a mistake in hurry, it should be `type.first.size()` not `type.size()`
haven't you locally compiled it before pushing the commit.
its a bad practice. first compile it localy if you made a change in code. 
if it works then push the change. If it isn't working after your changes and you are unable to detect , then you can push then up and ping collaborators to check why is it failing.

**Comment by srbcheema1** [comment]
Well done @hritikgupta the BUG is SHUT now. :)

I am merging this. looking forward to see more contributions from your side on more projects. especially I would love to see your contributions on https://github.com/srbcheema1/CheemaFy

**Pull request merged**